front/src/modules/simulationResult/helpers/configureHandlePan.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  isPointPickingElement,
+  isSegmentPickingElement,
+  type HoveredItem,
+  type SpaceTimeChartProps,
+} from '@osrd-project/ui-charts';
+import dayjs from 'dayjs';
+
+import type { TrainId } from 'reducers/osrdconf/types';
+import { updateSelectedTrainId } from 'reducers/simulationResults';
+import type { AppDispatch } from 'store';
+import {
+  extractOccurrenceIndexFromOccurrenceId,
+  extractPacedTrainIdFromOccurrenceId,
+  isOccurrenceId,
+  isTrainId,
+} from 'utils/trainId';
+
+import type { IndividualTrainProjection, TrainSpaceTimeData } from '../types';
+
+type DraggingState =
+  | {
+      draggedTrain: IndividualTrainProjection;
+      initialDepartureTime: Date;
+    }
+  | undefined;
+
+type ConfigureHandlePanParams = {
+  spaceTimeChartOnPan?: SpaceTimeChartProps['onPan'];
+  handleTrainDrag?: (args: {
+    draggedTrainId: TrainId;
+    initialDepartureTime: Date;
+    newDepartureTime: Date;
+    stopPanning: boolean;
+  }) => Promise<void>;
+  draggingState: DraggingState;
+  setDraggingState: (s: DraggingState) => void;
+  hoveredItem: HoveredItem | null;
+  previousPanning: boolean;
+  setPreviousPanning: (v: boolean) => void;
+  zoomMode: boolean;
+  projectPathTrainResult: TrainSpaceTimeData[];
+  dispatch: AppDispatch;
+};
+
+export function configureHandlePan({
+  spaceTimeChartOnPan,
+  handleTrainDrag,
+  draggingState,
+  setDraggingState,
+  hoveredItem,
+  previousPanning,
+  setPreviousPanning,
+  zoomMode,
+  projectPathTrainResult,
+  dispatch,
+}: ConfigureHandlePanParams): SpaceTimeChartProps['onPan'] {
```

Comment:
In `SpaceTimeChartProps`, `onPan` can be nullable, but not in your function :) So you can force the type by adding `NonNullable` and remove the noop function in `SpaceTimeChartWrapper`
```suggestion
}: ConfigureHandlePanParams): NonNullable<SpaceTimeChartProps['onPan']> {
```